duckduckgo/Android
# Tweak survey infrastructure

Task/Issue URL: https://app.asana.com/0/361428290920652/1198222027640374

**Description**:
Some small tweaks to our survey infrastructure
* only show surveys without a set day to installers who have been with us for more than 30 days
* restrict surveys to US locale
* fix issue where CTA could still show after survey completed
* small UI tweaks

**Steps to test this PR**:

**_Setup_**
1. Open SurveyService and update to our sandbox endpoint "http://ddg-sandbox.s3.amazonaws.com/survey/survey-mobile.json"

**_Test restrictions_**
1. Run a fresh install on a non-US locale and ensure that the survey CTA does not show on the home screen (minimum installation days needs to be 30 and locale must be US).
1. Open CTAViewModel and change line 248 to be `val daysInstalled = 30L`, run the app again and ensure that the survey does show on the home screen (locale needs to be US).
1. Change locale to US.
1. Run the app again, the survey should now show.

**_Test survey is dismissed when done_**
1. Run through the survey completing it.
1. Ensure that the CTA disappears from the home screen.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
